deploy/base/deployment.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: objectstorage-provisioner
+  namespace: objectstorage-provisioner-ns
+  labels:
+    app.kubernetes.io/part-of: container-object-storage-interface
+    app.kubernetes.io/component: provisioner
+    app.kubernetes.io/version: main
+    app.kubernetes.io/name: container-object-storage-interface-provisioner-sidecar
+spec:
+  replicas: 1
+  minReadySeconds: 30
+  progressDeadlineSeconds: 600
+  revisionHistoryLimit: 3
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
```

Comment:
Are these any different from the defaults? Are they required to be set to these values?

Reply:
Yes these values are different from defaults. By default `maxUnavailable` is `25%`. Since we are only running with single pod here, we should not terminate the old pod until the new pod is ready.